main.tf

```diff
@@ -3,6 +3,39 @@ locals {
 
   # normalize user input by appending trailing slash
   s3_exported_prefix = var.s3_exported_prefix != "" ? format("%s/", trimsuffix(var.s3_exported_prefix, "/")) : ""
+
+  default_actions = [
```

Comment:
This looks good, I'm just concerned that there's no simple way of getting all the supported actions without copy pasting a large list into `include`. That can be quite useful for testing for example.

You can add a `nullable` variable with this hardcoded list:
```
variable "snapshot_action" {
  description = "List of actions triggered by snapshot. Set to null to inherit all actions supported by the lambda."
  type        = set(string)
  nullable    = true
  default     = [ ... ]
}
```
And then have
```
module "observe_lambda_snapshot" {
  action = var.snapshot_action
  ...
}
```

Since `action` is not nullable, the effect of passing in a `null` is that we use the underlying default in the [snapshot module](https://github.com/observeinc/terraform-aws-lambda/blob/main/modules/snapshot/variables.tf#L32-L41)